test/libyul/YulOptimizerTestCommon.cpp

```diff
@@ -0,0 +1,467 @@
+	{
+		disambiguate();
+		if (m_fuzzerMode)
+			ForLoopInitRewriter::run(*m_context, *m_ast);
+		// apply SSA
+		SSATransform::run(*m_context, *m_ast);
+		RedundantAssignEliminator::run(*m_context, *m_ast);
+		// reverse SSA
+		SSAReverser::run(*m_context, *m_ast);
+		if (m_fuzzerMode)
+			FunctionHoister::run(*m_context, *m_ast);
+		CommonSubexpressionEliminator::run(*m_context, *m_ast);
+		UnusedPruner::run(*m_context, *m_ast);
+	}
+	else if (m_optimizerStep == "stackCompressor")
+	{
+		disambiguate();
+		if (m_fuzzerMode)
+		{
+			ForLoopInitRewriter::run(*m_context, *m_ast);
+			FunctionHoister::run(*m_context, *m_ast);
+		}
+		FunctionGrouper::run(*m_context, *m_ast);
+		size_t maxIterations = 16;
+		StackCompressor::run(*m_dialect, *m_object, true, maxIterations);
+		BlockFlattener::run(*m_context, *m_ast);
+	}
+	else if (m_optimizerStep == "wordSizeTransform")
+	{
+		disambiguate();
+		ExpressionSplitter::run(*m_context, *m_ast);
+		WordSizeTransform::run(*m_dialect, *m_dialect, *m_ast, *m_nameDispenser);
+	}
+	else if (m_optimizerStep == "fullSuite")
+	{
+		GasMeter meter(dynamic_cast<EVMDialect const&>(*m_dialect), false, 200);
+		OptimiserSuite::run(*m_dialect, &meter, *m_object, true, solidity::frontend::OptimiserSettings::DefaultYulOptimiserSteps);
+	}
+	else if (m_optimizerStep == "stackLimitEvader")
+	{
+		disambiguate();
+		StackLimitEvader::run(*m_context, *m_object, CompilabilityChecker{
+			*m_dialect,
+			*m_object,
+			true
+		}.unreachableVariables);
+	}
+	else if (m_optimizerStep == "fakeStackLimitEvader")
+	{
+		disambiguate();
+		// Mark all variables with a name starting with "$" for escalation to memory.
+		struct FakeUnreachableGenerator: ASTWalker
+		{
+			map<YulString, set<YulString>> fakeUnreachables;
+			using ASTWalker::operator();
+			void operator()(FunctionDefinition const& _function) override
+			{
+				YulString originalFunctionName = m_currentFunction;
+				m_currentFunction = _function.name;
+				ASTWalker::operator()(_function);
+				m_currentFunction = originalFunctionName;
+			}
+			void visitVariableName(YulString _var)
+			{
+				if (!_var.empty() && _var.str().front() == '$')
+					fakeUnreachables[m_currentFunction].insert(_var);
+			}
+			void operator()(VariableDeclaration const& _varDecl) override
+			{
+				for (auto const& var: _varDecl.variables)
+					visitVariableName(var.name);
+				ASTWalker::operator()(_varDecl);
+			}
+			void operator()(Identifier const& _identifier) override
+			{
+				visitVariableName(_identifier.name);
+				ASTWalker::operator()(_identifier);
+			}
+			YulString m_currentFunction = YulString{};
+		};
+		FakeUnreachableGenerator fakeUnreachableGenerator;
+		fakeUnreachableGenerator(*m_ast);
+		StackLimitEvader::run(*m_context, *m_object, fakeUnreachableGenerator.fakeUnreachables);
+	}
+	else
+		return false;
+
+	return true;
+}
+
+shared_ptr<Block> YulOptimizerTester::run()
+{
+	bool success = runStep();
+	if (!success)
+		return nullptr;
```

Comment:
```suggestion
	return nullptr;
```
Too much indentation?

Reply:
Hmm... I just checked locally that it's a single tab. Do you think it's additional spaces?

Reply:
Ah nevermind, I thought it was too long and copied it incorrectly while checking.